cleanlab/multiannotator.py

```diff
@@ -275,26 +300,439 @@ def get_label_quality_multiannotator(
                 annotator_agreement,
             )
 
-    if return_detailed_quality and return_annotator_stats:
-        return {
-            "label_quality": label_quality,
-            "detailed_label_quality": detailed_label_quality,
-            "annotator_stats": annotator_stats,
-        }
-    elif return_detailed_quality:
-        return {
-            "label_quality": label_quality,
-            "detailed_label_quality": detailed_label_quality,
-        }
-    elif return_annotator_stats:
-        return {
-            "label_quality": label_quality,
-            "annotator_stats": annotator_stats,
-        }
+    labels_info = {
+        "label_quality": label_quality,
+    }
+
+    if return_detailed_quality:
+        labels_info["detailed_label_quality"] = detailed_label_quality
+    if return_annotator_stats:
+        labels_info["annotator_stats"] = annotator_stats
+    if return_weights:
+        labels_info["model_weight"] = model_weight
+        labels_info["annotator_weight"] = annotator_weight
+
+    return labels_info
+
+
+def get_label_quality_multiannotator_ensemble(
+    labels_multiannotator: Union[pd.DataFrame, np.ndarray],
+    pred_probs: np.ndarray,
+    *,
+    temp_scale: bool = False,
+    return_detailed_quality: bool = True,
+    return_annotator_stats: bool = True,
+    return_weights: bool = False,
+    verbose: bool = True,
+    label_quality_score_kwargs: dict = {},
+) -> Dict[str, Any]:
+    """Returns label quality scores for each example and for each annotator.
+
+    This function is for multiclass classification datasets where examples have been labeled by
+    multiple annotators (not necessarily the same number of annotators per example).
+
+    It computes one consensus label for each example that best accounts for the labels chosen by each
+    annotator (and their quality), as well as a score for how confident we are that this consensus label is actually correct.
+    It also computes the the quality scores for each annotator's individual labels, and the quality of each annotator.
+    This function allows for an ensemble of models to be trained, the models' predictions and confidences will be used to compute the
+    consensus label and annotator quality.
```

Comment:
```suggestion
    This function is similar to :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>` but for settings where you have trained an ensemble of multiple classifier models rather than a single model.
    For multiclass classification datasets where examples have been labeled by
    multiple annotators, this function computes: a consensus label for each example, a score for how confident we are that this consensus label is actually correct, and quality scores for each annotator's individual labels as well as the quality of each annotator.
    Assuming an ensemble of multiple classifier models has been trained, all of the models' predictions and confidences are used to estimate consensus label and annotator quality in an aggregated fashion that can produce more accurate estimates than only training a single classifier model.
```